controller/pkg/usertokens/common/common.go

```diff
@@ -9,21 +14,67 @@ const (
 	OIDC
 )
 
-// FlattenClaim flattes all the generic claims in a flat array for strings.
+// convert an unknown int type to int64
+func toInt64(i interface{}) int64 {
+	switch i := i.(type) {
+	case int:
+		return int64(i)
+	case int8:
+		return int64(i)
+	case int16:
+		return int64(i)
+	case int32:
+		return int64(i)
+	case int64:
+		return i
+	default:
+		panic("toInt64(): expected a signed integer type, got " + reflect.TypeOf(i).String() + " instead")
+	}
+}
+
+// convert an unknown int type to int64
+func toUint64(i interface{}) uint64 {
+	switch i := i.(type) {
+	case uint:
+		return uint64(i)
+	case uint8:
+		return uint64(i)
+	case uint16:
+		return uint64(i)
+	case uint32:
+		return uint64(i)
+	case uint64:
+		return i
+	default:
+		panic("toUint64(): expected an unsigned integer type, got " + reflect.TypeOf(i).String() + " instead")
+	}
+}
+
+// FlattenClaim flattens all the generic claims in a flat array for strings.
 func FlattenClaim(key string, claim interface{}) []string {
 	attributes := []string{}
-	if slice, ok := claim.([]string); ok {
-		for _, data := range slice {
+
+	switch claim := claim.(type) {
+	case bool:
+		attributes = append(attributes, key+"="+strconv.FormatBool(claim))
+	case int, int8, int16, int32, int64:
+		attributes = append(attributes, key+"="+strconv.FormatInt(toInt64(claim), 10))
+	case uint, uint8, uint16, uint32, uint64:
+		attributes = append(attributes, key+"="+strconv.FormatUint(toUint64(claim), 10))
+	case float32:
+		attributes = append(attributes, key+"="+strconv.FormatFloat(float64(claim), 'G', -1, 32))
+	case float64:
+		attributes = append(attributes, key+"="+strconv.FormatFloat(claim, 'G', -1, 64))
+	case string:
+		attributes = append(attributes, key+"="+claim)
+	case []string:
+		for _, data := range claim {
```

Comment:
as discussed, a slice will most likely unmarshall to []interface{} .. so most likely the best way is call again the flattenclaims but with the slice array. This is the reason that []string itself doesn't work some times.

Reply:
This commit addresses the case when []interface{} :  7308a498